perf/perf_hv_gpci_interface.py

```diff
@@ -0,0 +1,171 @@
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+        self.expected_files = self.params.get("access_files", default=None)
+        self.hmc_ip = self.get_mcp_component("HMCIPAddr")
+        if not self.hmc_ip:
+            self.cancel("HMC IP not got")
+        self.hmc_username = self.params.get("hmc_username", '*', default=None)
+        self.hmc_pwd = self.params.get("hmc_pwd", '*', default=None)
+        self.lpar = self.get_partition_name("Partition Name")
+        if not self.lpar:
+            self.cancel("LPAR Name not got from lparstat command")
+        self.session = Session(self.hmc_ip, user=self.hmc_username,
+                               password=self.hmc_pwd)
+        if not self.session.connect():
+            self.cancel("failed connecting to HMC")
+        cmd = 'lssyscfg -r sys -F name'
+        output = self.session.cmd(cmd).stdout_text
+        self.server = ''
+        for line in output.splitlines():
+            if line in self.lpar:
+                self.server = line
+        if not self.server:
+            self.cancel("Managed System not got")
+
+        performance_cmd = f"lssyscfg -m {self.server} -r lpar --filter lpar_names={self.lpar} -F allow_perf_collection"
+        output = self.session.cmd(performance_cmd).stdout_text.strip()
+        if output != '1':
+            self.cancel("Performance property not enabled. Test cannot continue")
+
+        # create temporary user
+        if process.system('useradd test', sudo=True, ignore_status=True):
+            self.log.warn('test useradd failed')
+
+        # Clear the dmesg, by that we can capture the delta at the end of
+        # the test.
+        process.run("dmesg -C")
+
+    @staticmethod
+    def get_mcp_component(component):
+        '''
+        probes IBM.MCP class for mentioned component and returns it.
+        '''
+        for line in process.system_output('lsrsrc IBM.MCP %s' % component,
+                                          ignore_status=True, shell=True,
+                                          sudo=True).decode("utf-8") \
+                                                    .splitlines():
+            if component in line:
+                return line.split()[-1].strip('{}\"')
+        return ''
+
+    @staticmethod
+    def get_partition_name(component):
+        '''
+        get partition name from lparstat -i
+        '''
+
+        for line in process.system_output('lparstat -i', ignore_status=True,
+                                          shell=True,
+                                          sudo=True).decode("utf-8") \
+                                                    .splitlines():
+            if component in line:
+                return line.split(':')[-1].strip()
+        return ''
+
+    def run_cmd(self, cmd, user=None):
+        if user == 'test':  # i.e., we are running as root
```

Comment:
can you correct the comment for test user

Reply:
done